lib/livebook/hubs.ex

```diff
@@ -211,21 +211,45 @@ defmodule Livebook.Hubs do
   @doc """
   Creates a secret for given hub.
   """
-  @spec create_secret(Secret.t()) :: :ok | {:error, list({String.t(), list(String.t())})}
+  @spec create_secret(Secret.t()) :: :ok | {:error, list({atom(), list(String.t())})}
   def create_secret(%Secret{origin: {:hub, id}} = secret) do
-    case get_hub(id) do
-      {:ok, hub} ->
-        if capability?(hub, [:secrets]) do
-          Provider.create_secret(hub, secret)
-        else
-          {:error, Secrets.add_secret_error(secret, :origin, "is invalid")}
-        end
-
-      :error ->
-        {:error, Secrets.add_secret_error(secret, :origin, "is invalid")}
+    with {:ok, hub} <- get_hub(id),
+         true <- capability?(hub, [:create_secret]) do
+      Provider.create_secret(hub, secret)
+    else
+      _ -> {:error, add_invalid_origin_error(secret)}
```

Comment:
Maybe this should raise as in we should never call this function in the first place if we don't have the capability? So I mean something like this:

```suggestion
    {:ok, hub} = get_hub(id)
    true = capability?(hub, [:create_secret])
    Provider.create_secret(hub, secret)
```

Reply:
Same below.

Reply:
Yeah, makes sense, since we have the capabilities guard.